src/sentry/search/snuba/executors.py

```diff
@@ -93,112 +89,167 @@ def query(
         date_from,
         date_to,
     ):
+        """This function runs your actual query and returns the results
+        We usually return a paginator object, which contains the results and the number of hits"""
         raise NotImplementedError
 
+    def snuba_search(
+        self,
+        start,
+        end,
+        project_ids,
+        environment_ids,
+        sort_field,
+        cursor=None,
+        group_ids=None,
+        limit=None,
+        offset=0,
+        get_sample=False,
+        search_filters=None,
+    ):
+        """
+        Returns a tuple of:
+        * a sorted list of (group_id, group_score) tuples sorted descending by score,
+        * the count of total results (rows) available for this query.
+        """
 
-def snuba_search(
-    start,
-    end,
-    project_ids,
-    environment_ids,
-    sort_field,
-    cursor=None,
-    candidate_ids=None,
-    limit=None,
-    offset=0,
-    get_sample=False,
-    search_filters=None,
-):
-    """
-    This function doesn't strictly benefit from or require being pulled out of the main
-    query method above, but the query method is already large and this function at least
-    extracts most of the Snuba-specific logic.
-
-    Returns a tuple of:
-     * a sorted list of (group_id, group_score) tuples sorted descending by score,
-     * the count of total results (rows) available for this query.
-    """
-    filters = {"project_id": project_ids}
+        filters = {"project_id": project_ids}
 
-    if environment_ids is not None:
-        filters["environment"] = environment_ids
+        if environment_ids is not None:
+            filters["environment"] = environment_ids
 
-    if candidate_ids:
-        filters["issue"] = sorted(candidate_ids)
+        if group_ids:
+            filters[self.TABLE_ALIAS + "issue"] = sorted(group_ids)
```

Comment:
I believe this PR made a change here
https://github.com/getsentry/sentry/pull/15922/files#diff-e150078e0815328e046861a0214b1aab